sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/util/ClientLogger.java

```diff
@@ -149,7 +146,8 @@ public void log(LogLevel logLevel, Supplier<String> message) {
      */
     public void log(LogLevel logLevel, Supplier<String> message, Throwable throwable) {
```

Comment:
I wonder if we can remove these overloads.

They don't seem to provide any convenience and we don't really want anyone to format messages

I.e. instead of 
```java
logger.log(INFO, () -> String.format("hello %s", a), ex)
```

we'd rather want sdks to do

```java
logger.atLevel(INFO)
    .addKeyValuePair("a", a)
    .log("hello");
```

This would result in structured log and promote writing them instead of formatted messages.